disk-mapper/internal/rejoinclient/rejoinclient.go

```diff
@@ -130,19 +133,37 @@ func (c *RejoinClient) requestRejoinTicket(endpoint string) (*joinproto.IssueRej
 	return joinproto.NewAPIClient(conn).IssueRejoinTicket(ctx, &joinproto.IssueRejoinTicketRequest{DiskUuid: c.diskUUID})
 }
 
-// getControlPlaneEndpoints requests the available control-plane endpoints from the metadata API.
+// getJoinEndpoints requests the available control-plane endpoints from the metadata API.
 // The list is filtered to remove *this* node if it is a restarting control-plane node.
-func (c *RejoinClient) getControlPlaneEndpoints() ([]string, error) {
+// Furthermore, the load balancer's endpoint is added.
+func (c *RejoinClient) getJoinEndpoints() ([]string, error) {
 	ctx, cancel := c.timeoutCtx()
 	defer cancel()
-	endpoints, err := metadata.JoinServiceEndpoints(ctx, c.metadataAPI)
+	instances, err := c.metadataAPI.List(ctx)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("retrieving instances list from cloud provider: %w", err)
+	}
+
+	joinEndpoints := []string{}
+	for _, instance := range instances {
+		if instance.Role == role.ControlPlane {
+			if instance.VPCIP != "" {
+				joinEndpoints = append(joinEndpoints, net.JoinHostPort(instance.VPCIP, strconv.Itoa(constants.JoinServiceNodePort)))
+			}
+		}
 	}
+
+	lbEndpoint, _, err := c.metadataAPI.GetLoadBalancerEndpoint(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving load balancer endpoint from cloud provider: %w", err)
+	}
+	joinEndpoints = append(joinEndpoints, net.JoinHostPort(lbEndpoint, strconv.Itoa(constants.JoinServiceNodePort)))
```

Comment:
Same as in `joinclient.go`, this adds the LB as the last entry, making it the fallback.